version: '3.9'
services:
  frontend:
    build: ./frontend
    networks:
      - frontend_nw
    depends_on:
      - flask
    volumes:
      - ./frontend:/frontend

  flask:
    build: ./backend
    restart: on-failure
    stop_signal: SIGINT
    volumes:
      - ./backend:/backend
    networks:
      - backend_nw
    depends_on:
      - phone-db
    command: gunicorn -b flask:5000 app:app

  webserver:
    image: nginx:latest
    build: ./nginx
    restart: always
    volumes:
      - ./nginx/conf.d/app.conf:/etc/nginx/nginx.conf
    ports:
      - "8181:80"
    depends_on:
      - frontend
      - flask
    networks:
      - frontend_nw
      - backend_nw

  phone-db:
    image: postgres:alpine
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: postgres
    networks:
      - backend_nw
    healthcheck:
      test: [ "CMD", "pg_isready", "-U", "postgres" ]
      interval: 5s
      retries: 5
    volumes:
      - ./db/init.sql:/docker-entrypoint-initdb.d/init.sql
      - ./db/data/MNP.csv:/MNP.csv
      - ./db/data/Numbers-Plan-9.csv:/Numbers-Plan-9.csv

      - ./db/postgres-data:/var/lib/postgresql/data

networks:
  frontend_nw:
  backend_nw: